atc/creds/ssm/manager_factory.go

```diff
@@ -26,6 +27,17 @@ func (factory *ssmManagerFactory) AddConfig(group *flags.Group) creds.Manager {
 	return manager
 }
 
-func (factory *ssmManagerFactory) NewInstance(interface{}) (creds.Manager, error) {
-	return &SsmManager{}, nil
+func (factory *ssmManagerFactory) NewInstance(config interface{}) (creds.Manager, error) {
+	if c, ok := config.(map[string]interface{}); !ok {
```

Comment:
I think this check can probably just be removed and we can let the mapstructure call fail since the error message it returns will probably be more helpful anyway.

I think this was probably you copied from because we used to manually go over the fields, but now we just use `mapstructure` which makes the extra typecast redundant.

At that point `Config` could probably just be inlined since it's pretty much just doing one thing and is only ever used in one place. The full code would end up looking something like this:

```go
manager := &SsmManager{
  TeamSecretTemplate: DefaultTeamSecretTemplate,
  PipelineSecretTemplate : DefaultPipelineSecretTemplate,
}

err := // ... do the mapstructure stuff...
if err != nil {
  return nil, err
}

return manager, nil
```

Reply:
I've changed it, let me know if the changes I made is what you are looking for